check_outputs.py

```diff
@@ -1,39 +1,113 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ----------------------------------------------------------------------------
+import os
+import sys
+import h5py
 import pickle
+import logging
+import argparse
+import numpy as np
+import matplotlib.pyplot as plt
 
-# load the inference outputs from the pickle file
-with open('outputs.pkl', 'r') as fid:
-     (valout, testout) = pickle.load(fid)
+from neon import NervanaObject
+from neon.backends import gen_backend
+from neon.models import Model
+from neon.data import ImageParams
+from neon.models import Model
+from neon.util.persist import load_obj
+from neon.util.modeldesc import ModelDescription
 
-# open the h5 files with the validation and test images
-import os
-import h5py
-pth_to_camvid = 'SegNet-Tutorial/CamVid'  # or wherever the image hdf5 files were generated
-with h5py.File(os.path.join(pth_to_camvid, 'val_images.h5'), 'r') as fid:
-    shp = list(fid['input'].attrs['lshape'])
-    mns = fid['input'].attrs['mean']
-    shp.insert(0, -1)
-    # reshape to be (N, H, W, C)
-    val_imgs = np.array(fid['input']).reshape(shp).transpose((0,2,3,1))
+from segnet_neon_backend import NervanaGPU_Upsample
+from segnet_neon_backend import _get_bprop_upsampling
+from pixelwise_dataloader import PixelWiseImageLoader
+from segnet_neon import gen_model
+global _get_bprop_upsampling
 
-    # add the mean value back in for each channel
-    for ind in range(3):
-        val_imgs[:,:,:,ind] += mns[ind]
+parser = argparse.ArgumentParser(description='compare output of trained model to ground truth')
+parser.add_argument('image_path', type=str, help='path to the CamVid data set in the SegNet-Tutorial '
+                                                 'GitHub repo')
+parser.add_argument('save_model_file', type=str, help='serialized model file')
+parser.add_argument("--num_classes", type=int, default=12, help="number of annotated classes")
+parser.add_argument("--height", type=int, default=256, help="image height")
+parser.add_argument("--width", type=int, default=512, help="image width")
+args = parser.parse_args()
 
+# check that image dimensions are powers of 2
+if((args.height & (args.height - 1)) != 0):
+    raise TypeError("Height must be a power of 2.")
+if((args.width & (args.width - 1)) != 0):
+    raise TypeError("Width must be a power of 2.")
 
-import numpy as np
-import matplotlib.pyplot as plt
+(c, h, w) = (args.num_classes, args.height, args.width)
+
+# need to use the backend with the new upsampling layer implementation
+be = NervanaGPU_Upsample(device_id=0)
 
-# plot the first input image
-plt.imshow(val_imgs[0]/256)
+# set batch size
+be.bsz = 1
 
-# get the pixelwise classification from the softmax SegNet outputs
-img_segment = np.argmax(valout[0].reshape((12, 256, 512)), axis=0)
+# couple backend to global neon object
+NervanaObject.be = be
+
+shape = dict(channel_count=c, height=h, width=w, subtract_mean=False)
```

Comment:
dataloader will only scale the smaller dimension, i need the image to be 256x512 so since the camped is not a 1:2 ratio i don't think we can let data loader do this